src/main/java/com/amazon/sqs/javamessaging/AmazonSQSExtendedClient.java

```diff
@@ -775,15 +773,41 @@ public SendMessageBatchResult sendMessageBatch(SendMessageBatchRequest sendMessa
 
 		List<SendMessageBatchRequestEntry> batchEntries = sendMessageBatchRequest.getEntries();
 
+		List<Integer> smallMessages = new LinkedList<>();
```

Comment:
Now this is a two pass process. Move the obviously big ones to s3, catching the indices of those we didn't move.